### ISSUE TITLE
Property file with property expressions are flagged as wrong

### ISSUE THREAD

**Issue by gian1200** [eclipse/lsp4mp#225]
### Issue
Property definition is flagged when is assigned an ENV reference with a default value.

It fails with `quarkus.log.level`

![image](https://user-images.githubusercontent.com/405347/159553939-a0bdd61c-4922-4361-b91d-c37cdf2a0452.png)

This appears with Quick Fix:
![image](https://user-images.githubusercontent.com/405347/159554173-1ca44642-8654-46c5-a780-2cf0699ef173.png)

### Extensions version
- VSCode quarkus extension version: 1.10.0
![image](https://user-images.githubusercontent.com/405347/161294480-6af1d7c3-7116-4f71-968b-9567fa95ce77.png)

- Tools for MicroProfile version: 0.4.0
![image](https://user-images.githubusercontent.com/405347/161294440-c991af38-8445-4f3b-a67c-52ddf51aa847.png)

### Related Issues:

- https://github.com/redhat-developer/vscode-quarkus/issues/392
- https://github.com/eclipse/lsp4mp/issues/201

**Comment by angelozerr** [comment]
I can fix the first usecase with default value.

For the second usecase without default value
I dont know how to fix because it requires to find env variable which can be defined with several mean.

The mean that I could manage is to search in the .env file otherwise I dont know how to do that. We already discussion with @fbricon and @rgrunber of this problem and providing à settings to declare env variables is not à very good idea.

We could too provide a settings to ignore this error kind but with expression you can référence other properties so it can be helpful to have this validation.

Any feedback are welcome

**Comment by gian1200** [comment]
In our scenario we don't use an .env file.
- For running the app in Kubernetes we reference them using `quarkus.kubernetes.env.mapping`. Can the validation also consider those ENV defined at that level?
Something like:
![image](https://user-images.githubusercontent.com/405347/161307809-a72fbcdc-cb89-41b6-ae41-daee6ab462eb.png)

In addition to that, what do you think about flagging those variables that are not defined/found, but have a correct syntax, as warning instead of error. Maybe saying something like "Property/Variable xxxx could not be found locally. It should be defined at runtime to avoid unexpected issues", or something along the lines (maybe keep the "Unknown referenced property xxxx" message).

- For local testing we define those variables at SO level

In case the variable can be found in the developer's environment (System.getenv in java), then the warning might be omitted.

**Comment by angelozerr** [comment]
@gian1200 see the demo https://github.com/eclipse/lsp4mp/pull/224#issuecomment-1086904693 with the default value support.

Let me study more you suggestion https://github.com/eclipse/lsp4mp/issues/225#issuecomment-1086155352 to give you feedback.

**Comment by angelozerr** [comment]
> In addition to that, what do you think about flagging those variables that are not defined/found, but have a correct syntax, as warning instead of error. 

You can configure that, indeed there are settings https://github.com/redhat-developer/vscode-microprofile#supported-vs-code-settings  microprofile.tools.validation.*.severity per error code.

**Comment by angelozerr** [comment]
@gian1200 the default value has been fixed, please create a new issue with env variable problem without default value like you described in https://github.com/eclipse/lsp4mp/issues/225#issuecomment-1086155352

**Comment by gian1200** [comment]
> > In addition to that, what do you think about flagging those variables that are not defined/found, but have a correct syntax, as warning instead of error.
> 
> You can configure that, indeed there are settings https://github.com/redhat-developer/vscode-microprofile#supported-vs-code-settings microprofile.tools.validation.*.severity per error code.

Yes, that's exactly what I was describing! It even says in the docs that Warning is the default severiy for Unknown Properties. However, it doesn't seems to be working like that. Forcing `"microprofile.tools.validation.unknown.severity":"warning"` in `settings.json` does nothing. To make it work as expected, it looks like `"microprofile.tools.validation.expression.severity": "warning"` and `"microprofile.tools.validation.value.severity": "warning"` must be set, too. Will open a new issue to continue the conversation there.
![image](https://user-images.githubusercontent.com/405347/161657119-9fb78818-6de1-499f-86fa-c85cf37f530f.png)

Will create 2 tickets:
- ENV variable without default
- microprofile.tools.validation.unknown.severity warning is ignored

Thanks for your time!

PS: the demo looks great!

**Comment by angelozerr** [comment]
@gian1200 if you go to https://github.com/redhat-developer/vscode-microprofile#contributing you have à link to download last vsix. You can download and install it. You should have the feature of this issue.
